docs/guide/search/extend_elasticsearch.md

```diff
@@ -291,7 +291,210 @@
                 - { name: ezplatform.search.elasticsearch.query.location.sort_clause_visitor }
     ```
 
+    ## Custom Aggregation
+    
+    To create a custom aggregation for use with Elasticsearch, create an aggregation class.
+    In the following example, an aggregation groups Location query results according to the Location priority:
+    
+    ``` php
+    <?php
+
+    declare(strict_types=1);
+
+    namespace App\Query\Aggregation;
+
+    use eZ\Publish\API\Repository\Values\Content\Query\Aggregation\AbstractRangeAggregation;
+    use eZ\Publish\API\Repository\Values\Content\Query\Aggregation\LocationAggregation;
+
+    final class PriorityRangeAggregation extends AbstractRangeAggregation implements LocationAggregation
+    {
+
+    }
+    ```
+    
+    A custom aggregation requires an aggregation visitor and a result extractor.
```

Comment:
Here we also need explanation of what is aggregation visitor / result extractor as well as the explanation of its role and responsibilities.